### PR TITLE
feat: edit error/warning messages color of message logger

### LINKED ISSUE #521: Option to change the color of messages shown in the message logger
*I find red color irritating and difficult to read, unfortunately, by default all the variables printed in cerr are shown red in console. So it would be better if there is an option to select the color of messages*

### PR DESCRIPTION
**Pull request by brun0-matheus**
feat: Add option to choose error/warning message colors of message logger

## Description
Add two new settings: "Error Message Color" and "Warn Message Color".
These settings are located at Appearance -> General, and have a combo box to select the colors, using default [Qt Colors names](https://doc.qt.io/qt-5/stylesheet-reference.html#color). 
Message Logger code was modified to use these settings instead of hardcoded values.

## Related Issues / Pull Requests
Closes #521

## Motivation and Context
Some users don't like the default colors for various reasons, so, an option to change them was added.

## How Has This Been Tested?
Tested on Fedora Workstation 37, on dark and light modes.

## Checklist

- [x] If the key of a setting is changed, the `old` attribute is updated or it is resolved in SettingsUpdater.
- [x] If there are changes of the text displayed in the UI, they are wrapped in `tr()` or `QCoreApplication::translate()`.
- [ ] If needed, I have opened a pull request or an issue to update the [documentation](https://github.com/cpeditor/cpeditor.github.io).
- [ ] If these changes are notable, they are documented in [CHANGELOG.md](https://github.com/cpeditor/cpeditor/blob/master/CHANGELOG.md).

## Additional text
It's necessary to translate the settings description and also add in the documentation and changelog.

### CODE REVIEW

**Comment by coder3101** [comment]
Everything looks good!
However few suggestions before we merge this:
- Update the translations file. There is a script for that, see the action that failed above.
- Please check all of the combination of text colors and UI themes have enough contrast so people can read.
- MacOS static checks failed, you can fix it, although it's not related to your PR, looks like static checker got updated/improved.
- Update the changelog file.

**Comment by brun0-matheus** [comment]
About the contrast of text color and background, I've tested all the combinations, and most of them looks good to me, except for ones like dark text on dark background or white text on white background. The combinations that don't look good to me are:

For white background: lime, aqua, silver, yellow, white text color.
For black background: teal, purple, navy, maroon, black text color

**Comment by brun0-matheus** [comment]
The documentation pull request will be made soon. And about the mac os one, I will look into it in the near future (after docs).

**Comment by coder3101** [comment]
https://github.com/cpeditor/cpeditor/pull/1268 should unblock this PR, @brun0-matheus please rebase the changes on top of master.

File: CHANGELOG.md
Comment (coder3101) [inline CHANGELOG.md:6]:
```suggestion
-   Add options to select error/warning messages colors for message logger. (#1247 and #521)
```

**Review by ouuan — APPROVED** [review]

**Comment by coder3101** [comment]
@all-contributors please add @brun0-matheus  for code

**Pull request merged**
